src/BCReader.jl

```diff
@@ -211,21 +211,20 @@ function update_midmonth_data!(date, bcf_info::BCFileInfo{FT}) where {FT}
         bcf_info.monthly_fields[2] .= deepcopy(bcf_info.monthly_fields[1])
         bcf_info.segment_length .= Int(0)
 
-        # throw error when there are closer initial indices for the bc file data that matches this date0
-    elseif Dates.days(date - all_dates[Int(midmonth_idx + 1)]) > 2
+        # case 3: during initialization, warn and reset if there are closer file dates to the current date than the file date implied by midmonth_idx0+1
+    elseif (midmonth_idx == midmonth_idx0) && (date - all_dates[Int(midmonth_idx0) + 1]).value >= 0
         nearest_idx = argmin(
             abs.(
                 parse(FT, TimeManager.datetime_to_strdate(date)) .-
                 parse.(FT, TimeManager.datetime_to_strdate.(all_dates[:]))
             ),
         )
-        # TODO test this
-        bcf_info.segment_idx[1] = midmonth_idx = midmonth_idx0 = nearest_idx
+        bcf_info.segment_idx .= nearest_idx
+        bcf_info.segment_idx0 .= nearest_idx
         @warn "init data does not correspond to start date. Initializing with `SIC_info.segment_idx[1] = midmonth_idx = midmonth_idx0 = $nearest_idx` for this start date"
```

Comment:
Can we modify this warning to remove `midmonth_idx` and midmonth_idx0 since those are just local to this function? Also the message is hardcoded for SIC - maybe we could instead say "Initializing with `segment_idx = segment_idx0 = $nearest_idx` for this start date"

Reply:
Ooh, nice catch!